Components/SearchController.cs

```diff
@@ -95,45 +78,31 @@ private SearchController()
         /// <summary>
         /// Event handler for when the search text has been updated
         /// </summary>
-        public async Task UpdateSearchText(string searchText)
+        public List<SearchResult> UpdateSearchText(string searchText)
         {
             SearchText = searchText;
-            await SyncOpenWindowsWithModelAsync().ConfigureAwait(false);
+            return SyncOpenWindowsWithModel();
         }
 
         /// <summary>
         /// Syncs the open windows with the OpenWindows Model
         /// </summary>
-        public async Task SyncOpenWindowsWithModelAsync()
+        public List<SearchResult> SyncOpenWindowsWithModel()
         {
             System.Diagnostics.Debug.Print("Syncing WindowSearch result with OpenWindows Model");
 
             List<Window> snapshotOfOpenWindows = OpenWindows.Instance.Windows;
 
             if (string.IsNullOrWhiteSpace(SearchText))
             {
-                searchMatches = Main.Context.CurrentPluginMetadata.ActionKeyword != "*"?
-                                            snapshotOfOpenWindows.Select(x => new SearchResult { Result = x }).ToList():
+                return searchMatches = Main.Context.CurrentPluginMetadata.ActionKeyword != "*" ?
+                                            snapshotOfOpenWindows.Select(x => new SearchResult { Result = x }).ToList() :
                                             new List<SearchResult>();
             }
             else
```

Comment:
no big deal, just fyi if you are returning the result you dont need the else here.